src/main/java/reactor/ipc/netty/http/client/HttpClient.java

```diff
@@ -372,12 +373,13 @@ protected HttpClient(final HttpClientOptions options) {
 					} : EMPTY).onPipeline(this);
 		}
 
-		@Override
-		public void accept(ChannelPipeline pipeline, ContextHandler<Channel> c) {
-			pipeline.addLast(NettyPipeline.HttpDecoder, new HttpResponseDecoder())
-			        .addLast(NettyPipeline.HttpEncoder, new HttpRequestEncoder());
-		}
-	}
+        @Override
+        public void accept(ChannelPipeline pipeline, ContextHandler<Channel> c) {
+            pipeline.addLast(NettyPipeline.HttpDecompressor, new HttpContentDecompressor())
```

Comment:
Maybe that's where the API part takes place but it should be optional. Adding all the time the duplex handler weighs in stack traces and overhead